packages/web-ui/src/ds/molecules/MultiSelect/index.tsx

```diff
@@ -0,0 +1,362 @@
+// src/components/multi-select.tsx
+
+import * as React from 'react'
+
+import { cn } from '../../../lib/utils'
+import {
+  Badge,
+  Checkbox,
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+  FormField,
+  Popover,
+  Separator,
+} from '../../..'
+import { Button } from '../../ui/button'
+import Text from '../../atoms/Text'
+import { Icon } from '../../atoms/Icons'
+
+/**
+ * Props for MultiSelect component
+ */
+interface MultiSelectProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
+    Omit<typeof FormField, 'children'> {
+  /**
+   * An array of option objects to be displayed in the multi-select component.
+   * Each option object has a label, value, and an optional icon.
+   */
+  options: {
+    /** The text to display for the option. */
+    label: string
+    /** The unique value associated with the option. */
+    value: string
+    /** Optional icon component to display alongside the option. */
+    icon?: React.ComponentType<{ className?: string }>
+  }[]
+
+  /**
+   * Callback function triggered when the selected values change.
+   * Receives an array of the new selected values.
+   */
+  onChange?: (value: string[]) => void
```

Comment:
Make it as in select with a generic so the value here is what the options values are

Reply:
i think it's better if it's always a string since this way it behaves like a regular select that coerces all its values to strings

Reply:
Check Select I don't think it is necessary. It's working fine